src/mdlComponentHandler.js

```diff
@@ -342,7 +342,7 @@ componentHandler = (function() {
    * @param {?componentHandler.Component} component
    */
   function deconstructComponentInternal(component) {
-    var componentIndex = createdComponents_.indexOf(component);
+    var componentIndex = createdComponents_.indexOf(/** @type {componentHandler.Component} */ (component));
```

Comment:
cc @sgomes 

We need to type-hint the parameters now? If so this should probably get handled through another PR so it gets in faster.